sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubListener.PartitionProcessor.cs

```diff
@@ -287,19 +304,52 @@ private async Task MonitorCachedEvents(CancellationTokenSource backgroundCancell
                 }
             }
 
-            private TimeSpan GetRemainingTime(TimeSpan elapsed)
+            private TimeSpan GetAdjustedMaxWaitTime(TimeSpan configuredMaxWaitTime, DateTimeOffset? lastCheckpointTime, bool firstInvocation)
             {
-                if ((_maxWaitTime == Timeout.InfiniteTimeSpan) || (_maxWaitTime == TimeSpan.Zero) || (elapsed == TimeSpan.Zero))
+                // If this is not the first invocation of the function for this partition then use the configured wait time.
+                if (!firstInvocation)
+                {
+                    return configuredMaxWaitTime;
+                }
+
+                // If there was no checkpoint found, then it is safer to dispatch a partial batch for the
+                // first invocation than assume no ownership changes are taking place.
+                if (!lastCheckpointTime.HasValue)
+                {
+                    return TimeSpan.Zero;
+                }
+
+                var timeSinceLastCheckpoint = DateTimeOffset.UtcNow.Subtract(lastCheckpointTime.Value);
+
+                // If the checkpoint was last recorded more than the 1.5 times configured wait time ago, ownership
+                // of the partition likely migrated and the function was not invoked on the expected schedule.
+                // Dispatch any available events immediately, as the wait time is presumed to have already elapsed.
+                if (timeSinceLastCheckpoint.TotalMilliseconds >= (configuredMaxWaitTime.TotalMilliseconds * 1.5))
```

Comment:
Just curious why 1.5* the wait time instead of the wait time?

Reply:
This came out of our conversation.  The concern was that we may be too reactive when checkpointing is configured to be infrequent.   We had talked about doubling the max wait time, but that seemed like it may be too large - so I split the difference.  

Feeling different about padding the wait time now?  Should we consider snapping back to just the wait time?

Reply:
Ohh ok, I must have explained my thought wrong I was thinking
- if > max wait time but less than 2maxwait time then invoke immediately
- if > 2maxwaittime don't invoke immediately if less than minbatchsize, just start the timer and behave normally

I figured that a migration would take less than a full maxwaittime in most cases, so it would hopefully eliminate the case where a new app is starting up after awhile and the first batch is just small

Reply:
The concern that I have with the second clause there is that we don't know how frequently an application is checkpointing.  The migration may be fresh but the old owner is checkpointing every 10 batches.  Generally, I think we're better dispatching a partial batch for the first execution, since we know that it's a one-time thing and we'll be able to make more accurate decisions starting with the second batch.

Reply:
Yeah that makes sense, my thought was to try and avoid having the first small batch hit every application, but having the documentation right in the property that you're likely to get a small batch in the first invocation is a good compromise.

Reply:
ok.  I'm going to bump this back to the configured wait time and remove the multiplier while ignoring the second clause.